### ISSUE TITLE
[Bug] Expression substitution doesn't always work

### ISSUE THREAD

**Issue by roryabraham** [rhysd/actionlint#113]
From https://github.com/rhysd/actionlint/blob/main/docs/checks.md#shellcheck-integration-for-run:

> Since both ${{ }} expression syntax and ShellScript's variable access $FOO use $, remaining ${{ }} confuses shellcheck. To avoid it, actionlint replaces ${{ }} with underscores. For example echo '${{ matrix.os }}' is replaced with echo '________________'.

Unfortunately, I found an example where this doesn't work:

```yaml
run: |
  if [[ -z ${{ env.FOO }} ]]; then
    echo "FOO is empty"
  fi
```

**Comment by roryabraham** [comment]
Found a workaround but it's not ideal:

```yaml
run: |
  SOME_VAR=${{ env.SOME_VAR }}
  if [[ -z "$SOME_VAR" ]]; then
    echo "SOME_VAR is empty"
  fi
```

**Comment by rhysd** [comment]
Thank you for reporting this. Really nice catch. I will temporarily disable SC2157 then consider how to handle this correctly later.

**Comment by rhysd** [comment]
I added workaround at https://github.com/rhysd/actionlint/commit/29512f4501a99d903e001f458b4673f7f692bdc9. Closing.
